### ISSUE TITLE
[BC Break] Separate AST modification out

### ISSUE THREAD

**Issue by asgrim** [Roave/BetterReflection#182]
Features like #180 and #141 are methods that allow modification of the reflection (via manipulating the AST).

For 2.0 we should change the API so that the modification stuff is separated.

**Comment by asgrim** [comment]
Suggestion:

```php
<?php

// In current BR:
$reflectionMethod = $reflectionClass->getMethod('myMethod');
$reflectionMethod->removeReturnType();

// In 2.0
$newReflectionMethod = (new \BetterReflection\Mutation\RemoveReturnType())($reflectionMethod);
```

**Comment by Ocramius** [comment]
Partially implemented in #339. Closing here, as further discussion should be continued on the PR.
